src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -2975,12 +2982,13 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
     HWIntrinsicCategory category    = HWIntrinsicInfo::lookupCategory(intrinsicId);
     int                 numArgs     = HWIntrinsicInfo::lookupNumArgs(node);
     var_types           baseType    = node->gtSIMDBaseType;
+    unsigned            simdSize    = node->gtSIMDSize;
 
     GenTree* op1 = node->gtGetOp1();
     GenTree* op2 = node->gtGetOp2();
     GenTree* op3 = nullptr;
 
-    if (!HWIntrinsicInfo::SupportsContainment(intrinsicId))
+    if (!HWIntrinsicInfo::SupportsContainment(intrinsicId) || (simdSize == 8) || (simdSize == 12))
```

Comment:
We should be able to support containment on `simdSize == 12` if it is a local or one of the other cases we allocate 16-bytes of storage for it.

Reply:
Just noting, it is retyped but the `simdSize` isn't changed, and even if `node` is retyped, it doesn't mean `op1` or `op2` are retyped.
So it will require a few changes to get correct, but shouldn't be too difficult overall.